recipes/vtk/all/conanfile.py

```diff
@@ -0,0 +1,197 @@
+                "ioexport": [True, False], "mpi_minimal": [True, False]}
+    default_options = {"shared": False, "with_qt": False, "mpi": False, "fPIC": False,
+                "minimal": False, "ioxml": False, "ioexport": False, "mpi_minimal": False}
+    topics = ("conan", "VTK", "3D rendering", "2D plotting", "3D interaction", "3D manipulation", 
+                "graphics", "image processing", "scientific visualization", "geometry modeling")
+    short_paths = True
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Linux":
+            raise ConanInvalidConfiguration("VTK Conan package do not support Linux yet. Any help appreciated!")
+        if self.options.with_qt:
+            if self.options["qt"].shared == False:
+                raise ConanInvalidConfiguration("VTK option 'with_qt' requires 'qt:shared=True'")
+            if self.settings.os == "Linux":
+                if self.options["qt"].qtx11extras == False:
+                    raise ConanInvalidConfiguration("VTK option 'with_qt' requires 'qt:qtx11extras=True'")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("{}-{}".format(self.name, self.version), self.source_subfolder)
+
+    # Below code is inncorrect and should be removed after successful Lnx package build.
+    # Until then, keeping it for reference. 
+    # def _system_package_architecture(self):
+    #     if tools.os_info.with_apt:
+    #         if self.settings.arch == "x86":
+    #             return ':i386'
+    #         elif self.settings.arch == "x86_64":
+    #             return ':amd64'
+    #
+    #     if tools.os_info.with_yum:
+    #         if self.settings.arch == "x86":
+    #             return '.i686'
+    #         elif self.settings.arch == 'x86_64':
+    #             return '.x86_64'
+    #     return ""
+    #
+    # def build_requirements(self):
+    #     pack_names = None
+    #     if not self.options.minimal and tools.os_info.is_linux:
+    #         if tools.os_info.with_apt:
+    #             pack_names = [
+    #                 "freeglut3-dev",
+    #                 "mesa-common-dev",
+    #                 "mesa-utils-extra",
+    #                 "libgl1-mesa-dev",
+    #                 "libglapi-mesa",
+    #                 "libsm-dev",
+    #                 "libx11-dev",
+    #                 "libxext-dev",
+    #                 "libxt-dev",
+    #                 "libglu1-mesa-dev"]
+    #
+    #     if pack_names:
+    #         installer = tools.SystemPackageTool()
+    #         for item in pack_names:
+    #             installer.install(item + self._system_package_architecture())
+    
+    def requirements(self):
+        # 1) Is it correct that "opengl/system" is required for every OS?
+        # 2) Is it correct that "opengl/system" is even when compiled without Qt?
```

Comment:
I don't understand this question. Can you please clarify this?

Reply:
Refer to above comment